lib/sinatra/base.rb

```diff
@@ -920,7 +920,6 @@ def initialize(app = nil)
       super()
       @app = app
       @template_cache = Tilt::Cache.new
-      @pinned_response = nil # whether a before! filter pinned the content-type
```

Comment:
nit: You may want to leave this here. If I remember correctly, initializing instance variables at the stage of initialization is in line with ruby's way. This will show up a new warning like `warning: instance variable @pinned_response not initialized` if you keep this change. Of course, I'm recognizing that there are a lot of warnings similar to that already, though.